packages/core-manager/src/actions/snapshots-restore.ts

```diff
@@ -0,0 +1,52 @@
+import { Application, Container, Contracts } from "@arkecosystem/core-kernel";
+import { join } from "path";
+
+import { Actions } from "../contracts";
+import { Identifiers } from "../ioc";
+import { SnapshotsManager } from "../snapshots/snapshots-manager";
+
+@Container.injectable()
+export class Action implements Actions.Action {
+    public name = "snapshots.restore";
+
+    public schema = {
+        type: "object",
+        properties: {
+            name: {
+                type: "string",
+            },
+            truncate: {
+                type: "boolean",
+            },
+            verify: {
+                type: "boolean",
+            },
+        },
+    };
+
+    @Container.inject(Container.Identifiers.Application)
+    private readonly app!: Application;
+
+    @Container.inject(Identifiers.SnapshotsManager)
+    private readonly snapshotManager!: SnapshotsManager;
+
+    @Container.inject(Container.Identifiers.FilesystemService)
+    private readonly filesystem!: Contracts.Kernel.Filesystem;
+
+    public async execute(params: { name: string; truncate?: boolean; verify?: boolean }): Promise<any> {
+        const snapshotsDir = `${process.env.CORE_PATH_DATA}/snapshots/`;
+        const snapshotPath = join(snapshotsDir, params.name);
+
+        if (!(await this.filesystem.exists(snapshotPath))) {
+            throw new Error("Snapshot not found");
+        }
+
+        await this.snapshotManager.restore({
+            network: this.app.network(),
+            blocks: params.name,
+            ...params,
+        });
+
+        return {};
```

Comment:
This should respond with a `202` status code to indicate that the request was accepted but some processing is ongoing.

Reply:
202 is for HTTP response. Actions are part of the JSON-RPC specification and doesn't know nothing about HTTP protocol. We can still add responseCode parameter into action response that will override default 200 response code, but that requires changes in [RCP](https://github.com/faustbrian/hapi-json-rpc) plugin or addition plugin that will check the action responseCode parameter.